File: backend/services/admin/src/Enma.Admin.Application/Services/Super/SuperOrganizationsService.cs
using Enma.Admin.Application.Abstractions;
using Enma.Admin.Application.Contracts;
using Enma.Admin.Application.Dto.Super;
using Enma.Common.Models;
using FluentResults;

namespace Enma.Admin.Application.Services.Super;

internal sealed class SuperOrganizationsService : ISuperOrganizationsService
{
    private const int MaxPageSize = 200;

    private readonly IOrganizationsRepository _organizationsRepository;

    public SuperOrganizationsService(IOrganizationsRepository organizationsRepository)
    {
        _organizationsRepository = organizationsRepository;
    }

    public async Task<Result<PaginatedResult<SuperOrganizationListItemDto>>> ListAsync(
        int page,
        int pageSize,
        string? search,
        bool includeDeleted,
        CancellationToken ct = default)
    {
        if (page < 1) page = 1;
        if (pageSize < 1) pageSize = 20;
        if (pageSize > MaxPageSize) pageSize = MaxPageSize;

        var listRes = await _organizationsRepository.ListSuperAsync(page, pageSize, search, includeDeleted, ct);
        if (listRes.IsFailed) return Result.Fail<PaginatedResult<SuperOrganizationListItemDto>>(listRes.Errors);

        var countRes = await _organizationsRepository.CountSuperAsync(search, includeDeleted, ct);
        if (countRes.IsFailed) return Result.Fail<PaginatedResult<SuperOrganizationListItemDto>>(countRes.Errors);

        return PaginatedResult<SuperOrganizationListItemDto>.Create(listRes.Value, countRes.Value, page, pageSize);
    }

    public async Task<Result<SuperOrganizationDetailsDto>> GetByIdAsync(Guid organizationId, CancellationToken ct = default)
    {
        return await _organizationsRepository.GetSuperDetailsAsync(organizationId, ct);
    }
}

File: frontend/src/services/admin/super/superOrganizationsService.ts
import { apiClient } from "@/api/apiClient";
import type { Guid, PaginatedResult } from "@/types/admin.types";
import type {
  SuperListQuery,
  SuperOrganizationDetailsDto,
  SuperOrganizationListItemDto,
} from "@/types/super-admin.types";

// Future: softDelete, restore, transferOwnership.
export default class SuperOrganizationsService {
  private readonly baseUrl = "/api/admin/v1/super/organizations";

  public async list(
    query: SuperListQuery = {},
  ): Promise<PaginatedResult<SuperOrganizationListItemDto>> {
    const { data } = await apiClient.get<PaginatedResult<SuperOrganizationListItemDto>>(
      this.baseUrl,
      {
        params: {
          page: query.page ?? 1,
          pageSize: query.pageSize ?? 20,
          search: query.search || undefined,
          includeDeleted: query.includeDeleted ?? false,
        },
      },
    );
    return data;
  }

  public async getById(organizationId: Guid): Promise<SuperOrganizationDetailsDto> {
    const { data } = await apiClient.get<SuperOrganizationDetailsDto>(
      `${this.baseUrl}/${organizationId}`,
    );
    return data;
  }
}
